packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -368,7 +368,7 @@ abstract class ScrollPosition extends ViewportOffset with ScrollMetrics {
         notifyListeners();
         didUpdateScrollPositionBy(pixels - oldPixels);
       }
-      if (overscroll != 0.0) {
+      if (overscroll.abs() > precisionErrorTolerance) {
```

Comment:
Ah yes, this is a much clearer culprit. 👍  Thank you both for looking into this!